.github/workflows/.pull_request.yml

```
@@ -0,0 +1,96 @@
+# ==========================================================================
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+# ==========================================================================
+
+name: gcc/clang build and functional testing
+
+on:
+  pull_request:
+    branches: [ develop ]
+
+env:
+  # (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  ci:
+    # Latest is >= 20.04 which has required GNU >= 8.2
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        compiler: [ gcc, clang ]
+      # If one job fails, the other can carry on
+      fail-fast: false
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    # QPL requirement
+    - name: Install nasm
+      run: |
+        sudo apt-get update -y
+        sudo apt-get install -y nasm
+
+    - name: Setup clang
+      if: ${{ matrix.compiler == 'clang' }}
+      id: setup_clang
+      run: |
+        sudo apt-get update -y
+        sudo apt-get -y install clang-12
+        echo "::set-output name=c_comp::clang"
+        echo "::set-output name=cpp_comp::clang++"
+
+    - name: Configure cmake
+      env:
+        CC: ${{steps.setup_clang.outputs.c_comp}}
+        CXX: ${{steps.setup_clang.outputs.cpp_comp}}
+      run: cmake -B build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DCMAKE_INSTALL_PREFIX=installation
+
+    - name: Build
+      id: build
+      run: cmake --build build --target install
+
+    - name: Check on build failures
+      if: steps.build.outcome != 'success'
+      run: |
+        echo "Build step failed"
+        exit 1
+
+    - name: Algorithmic tests
+      id: ta_tests
+      continue-on-error: true
```

`continue-on-error: true` will always marks this step as success. I prefer to directly see if this step fails or not.
```suggestion
      if: ${{ always() && steps.build.outcome == 'success' }}
```